tests/MongoDB.Driver.Tests/PipelineDefinitionBuilderTests.cs

```diff
@@ -29,6 +29,20 @@ namespace MongoDB.Driver.Tests
     public class PipelineDefinitionBuilderTests
     {
         // public methods
+        [Fact]
+        public void Create_should_throw_exception_when_output_type_does_not_match_last_stage_type()
+        {
+            var finalStageDefinition = new JsonPipelineStageDefinition<BsonDocument, BsonDocument>("{ project: { } } ");
+
+            var exception = Record.Exception(() => PipelineDefinition<BsonDocument, Starship>.Create(
+                stages: new IPipelineStageDefinition[] { finalStageDefinition }));
+
+            var expectedException =
+                $"The output type to the last stage was expected to be {typeof(BsonDocument)}, but was {typeof(Starship)}.";
+            exception.Should().BeOfType<ArgumentException>();
+            exception.Message.Should().Contain(expectedException);
+        }
+
```

Comment:
Summary of above recommendations:

    [Fact]
    public void Create_should_throw_exception_when_last_stage_output_type_does_not_match_pipeline_output_type()
    {
        var finalStage = new JsonPipelineStageDefinition<BsonDocument, BsonDocument>("{ project : { } } ");
        var stages = new [] { finalStage };

        var exception = Record.Exception(() => PipelineDefinition<BsonDocument, object>.Create(stages));

        var e = exception.Should().BeOfType<ArgumentException>().Subject;
        e.ParamName.Should().Be("stages");
        e.Message.Should().Contain($"The output type to the last stage was expected to be {typeof(object)}, but was {typeof(BsonDocument)}.");
    }

Reply:
I'll leave my previous comment to illustrate the preferred style but since writing that I have realized that this test is in the wrong file and that there is an existing test that should be refactored/replaces in `PipelineDefinitionTests.cs`.

Reply:
I've refactored to use a test similar to the one in PipelineDefinitionTests :)